contracts/MechMarketplace.sol

```diff
@@ -381,132 +376,268 @@ contract MechMarketplace is IErrorsMarketplace {
-        MechDelivery storage mechDelivery = mapRequestIdDeliveries[requestId];
-        address priorityMech = mechDelivery.priorityMech;
+        uint256 numDeliveries;
+        uint256 numRequests = requestIds.length;
+        // Allocate delivered requests array
+        deliveredRequests = new bool[](numRequests);
+        // Allocate requester related arrays
+        address[] memory requesters = new address[](numRequests);
+        uint256[] memory requesterDeliveryRates = new uint256[](numRequests);
+
+        // Get delivery mech payment type
+        bytes32 paymentType = IMech(msg.sender).paymentType();
+
+        // Traverse all requests being delivered
+        for (uint256 i = 0; i < numRequests; ++i) {
+            // Get request info struct
+            RequestInfo storage requestInfo = mapRequestIdInfos[requestIds[i]];
+            address priorityMech = requestInfo.priorityMech;
+
+            // TODO continue instead of revert? This would let us just ignore incorrect requests without reverting the bulk
+            // TODO Additional thought: leave revert as it would stop from mech trying to request arbitrary requests and not actually recorded ones
+            // Check for request existence
+            if (priorityMech == address(0)) {
+                revert ZeroAddress();
+            }
 
-        // Check for request existence
-        if (priorityMech == address(0)) {
-            revert ZeroAddress();
-        }
+            // Check payment type
+            if (requestInfo.paymentType != paymentType) {
+                revert WrongPaymentType(paymentType);
+            }
 
-        // Check if request has been delivered
-        if (mechDelivery.deliveryMech != address(0)) {
-            revert AlreadyDelivered(requestId);
-        }
+            // Check if request has been delivered
+            if (requestInfo.deliveryMech != address(0)) {
+                continue;
+            }
 
-        // If delivery mech is different from the priority one
-        if (priorityMech != msg.sender) {
-            // Within the defined response time only a chosen priority mech is able to deliver
-            if (block.timestamp > mechDelivery.responseTimeout) {
-                // Decrease priority mech karma as the mech did not deliver
-                IKarma(karma).changeMechKarma(priorityMech, -1);
-                // Revoke request from the priority mech
-                IMech(priorityMech).revokeRequest(requestId);
-            } else {
-                // Priority mech responseTimeout is still >= block.timestamp
-                revert PriorityMechResponseTimeout(mechDelivery.responseTimeout, block.timestamp);
+            // TODO continue instead of revert? This would let us just ignore incorrect requests without reverting the bulk
+            // TODO Additional thought: The price between posting requests and delivering could have changed, thus just continue such that mech cleans those request info on their side
```

Comment:
Hm. need to understsnd the scenarios

Reply:
When user posted a request - the rate was 5. Before request is processed and delivered, the price was increased by the mech operator. Now mech wants a higher rate for a request, but the user has agreed on a smaller rate.